src/lib.rs

```diff
@@ -358,66 +359,72 @@ impl<'a> FMatcher<'a> {
+                                if self.options.distinct_name_matching {
+                                    for (x, y) in names.iter().chain(new_names.iter()) {
+                                        if x != &ptnm.as_str() && y == &textm.as_str() {
+                                            return false;
+                                        }
                                     }
                                 }
-                                Entry::Vacant(_) => match new_names.entry(ptnm.as_str()) {
+                                if textm.start() == textm.end() {
+                                    panic!("Text pattern matched the empty string.");
+                                }
+                                match names.entry(ptnm.as_str()) {
                                     Entry::Occupied(e) => {
                                         if e.get() != &textm.as_str() {
                                             return false;
                                         }
                                     }
-                                    Entry::Vacant(e) => {
-                                        e.insert(textm.as_str());
-                                    }
-                                },
+                                    Entry::Vacant(_) => match new_names.entry(ptnm.as_str()) {
+                                        Entry::Occupied(e) => {
+                                            if e.get() != &textm.as_str() {
+                                                return false;
+                                            }
+                                        }
+                                        Entry::Vacant(e) => {
+                                            e.insert(textm.as_str());
+                                        }
+                                    },
+                                }
+                                text = &text[textm.end()..];
+                            } else {
+                                return false;
                             }
-                            text = &text[textm.end()..];
-                        } else {
-                            return false;
                         }
                     }
-                    if (eww && text.starts_with(&ptn[..ptn.len() - WILDCARD.len()])) || ptn == text
-                    {
-                        names.extend(new_names);
-                        true
-                    } else {
-                        false
+                    if !matched {
+                        break;
                     }
                 }
-                None => {
-                    if eww {
-                        text.starts_with(&ptn[..ptn.len() - WILDCARD.len()])
-                    } else {
-                        ptn == text
-                    }
+                if (eww && text.starts_with(&ptn[..ptn.len() - WILDCARD.len()])) || ptn == text {
```

Comment:
I know this isn't a new variable name, but `eww` is indeed "eww". Should it be renamed (in another PR) as I can't guess what it could mean :P

Same for `sww`.

Reply:
We can argue over the names if you want (I think they're OK if you read the code), but I think we should declare that out-of-scope for this PR.

Reply:
Agreed that it's not for this PR, and probably pro-level bikeshedding too.

But honestly `eww` could have been called `dolphin_jimbler` and convey about the same meaning to the casual reader :P

Not a hill I will die on anyway.

Reply:
dolphin_jimbler? :p

Reply:
Exactly!